File: MVVM_android_java/app/src/main/java/com/example/mvvm_android_java/LoginViewModel.java
package com.example.mvvm_android_java;

import androidx.databinding.BaseObservable;
import androidx.databinding.Bindable;
import androidx.databinding.ObservableField;

public class LoginViewModel extends BaseObservable {

    private String email;
    private String password;
    public ObservableField<String>  messageLogin = new ObservableField<>();
    public ObservableField<Boolean> isShowMess = new ObservableField<>();
    public ObservableField<Boolean> colorSuccess = new ObservableField<>();

    @Bindable
    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
        notifyPropertyChanged(BR.email);
    }

    @Bindable
    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
        notifyPropertyChanged(BR.password);
    }

    public void onClickLogin(){
        isShowMess.set(true);
        User user =new User(getEmail(), getPassword());
        if (user.isValidEmail() && user.isPassword()){
            messageLogin.set("Login successful");
            colorSuccess.set(true);
        }else {
            messageLogin.set("Email or Password is not valid");
            colorSuccess.set(false);
        }
    }
}

File: login_MVVM_kotlin/app/src/main/java/com/example/login_mvvm_kotlin/LoginViewModel.kt
package com.example.login_mvvm_kotlin


import android.content.Intent
import androidx.core.content.ContextCompat.startActivity
import androidx.databinding.BaseObservable
import androidx.databinding.ObservableField


class LoginViewModel {
    var email: String = ""
    var password: String = ""
    val messageLogin = ObservableField<String>()
    var isShowMess = ObservableField<Boolean>()
    var colorSuccess = ObservableField<Boolean>()


    fun onClickLogin() {
        isShowMess.set(true)
        val user = User(email, password)
        if (user.isValidEmail() && user.isPassword()) {
            messageLogin.set("login successful")
            colorSuccess.set(true)
        } else {
            messageLogin.set("email or password is valid")
            colorSuccess.set(false)
        }
    }
}